server/src/main/java/org/elasticsearch/index/engine/CombinedDocValues.java

```diff
@@ -55,19 +56,18 @@ long docVersion(int segmentDocId) throws IOException {
     long docSeqNo(int segmentDocId) throws IOException {
         assert seqNoDV.docID() < segmentDocId;
         if (seqNoDV.advanceExact(segmentDocId) == false) {
+            assert false : "DocValues for field [" + SeqNoFieldMapper.NAME + "] is not found";
             throw new IllegalStateException("DocValues for field [" + SeqNoFieldMapper.NAME + "] is not found");
         }
         return seqNoDV.longValue();
     }
 
     long docPrimaryTerm(int segmentDocId) throws IOException {
-        if (primaryTermDV == null) {
-            return -1L;
-        }
         assert primaryTermDV.docID() < segmentDocId;
         // Use -1 for docs which don't have primary term. The caller considers those docs as nested docs.
```

Comment:
This comment looks in need of adjustment. Maybe it deserves a Javadoc comment to say it doesn't work with child docs too.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/51279/commits/5b7c5b5820b6d2182cc8f3fe0e6838e68e4163ca.